doc/user_guide/ml_benchmark.rst

```diff
@@ -0,0 +1,74 @@
+.. _ml_benchmark:
+
+Setting up an ML benchmark
+==========================
+
+Benchopt can be used to run benchmarks on machine learning problems. This page
+explains the specificities of setting up a benchmark in this context.
+
+Cross-validation
+----------------
+
+In order to evaluate the generalization performance of a method, a common
+practice is to use cross-validation. In this context, the data is split
+several times, and the model is trained on a subset of the data and evaluated
+on another independent subset. This process is repeated several times, and
+the average performance is reported.
+
+In benchopt, cross-validation is handled as separate runs of the ``Solver``,
+where the data is split into folds in ``Objective.get_objective``, by calling
+``Objective.get_split``. This method takes in the data to split (typically
+``numpy`` arrays or ``pandas`` dataframes) and returns the split data.
+The way the splits are defined depend on the ``Objective.cv`` attribute, which
+must be defined by the user. A typical workflow is the following:
+
+.. code-block:: python
+
+    class Objective(BaseObjective):
+        ...
+        def set_data(self, X, y):
+            self.X, self.y = X, y
+            # Specify a cross-validation splitter as the ``cv`` attribute.
+            # This will be automatically used in ``self.get_split`` to split
+            # the arrays provided.
+            self.cv = GroupKFold(n_splits=5, random_state=self.seed)
+
+            # If the cross-validation requires some metadata, it can be
+            # provided in the ``cv_metadata`` attribute. This will be passed
```

Comment:
To be explicit this will be passed to self.cv.split and self.cv.get_n_splits ?

Reply:
yes!